superset/db_engine_specs/base.py

```diff
@@ -486,14 +496,13 @@ def extract_error_message(cls, e: Exception) -> str:
         return f"{cls.engine} error: {cls._extract_error_message(e)}"
 
     @classmethod
-    def _extract_error_message(cls, e: Exception) -> str:
+    def _extract_error_message(cls, e: Exception) -> Optional[str]:
         """Extract error message for queries"""
         return utils.error_msg_from_exception(e)
 
     @classmethod
-    def adjust_database_uri(cls, uri, selected_schema: Optional[str]):
-        """Based on a URI and selected schema, return a new URI
-
+    def adjust_database_uri(cls, uri: URL, selected_schema: Optional[str]) -> None:
```

Comment:
Given that we are mutating the `uri` object in-place, there's no need to return the object.

Reply:
sometimes returning the object is nice so you can string together mutations on an object. Maybe we should keep returning it and type it appropriately?

Reply:
If memory serves me right, these wouldn't benefit from chaining in the context they're used. However, I don't oppose returning the mutated object, but in that case I'd add a note in the docstring. Also there's a `# TODO:` below that can be removed.